Failing Windows subscriptions test

Not sure whether it's a bug, but these tests were OK several days ago and it's better to be fixed.
```
2023-03-29T03:43:39.5184041Z === CONT  TestWSClient_SubscriptionsCompat/deprecated,_filtered
2023-03-29T03:43:39.5184199Z     client_test.go:2415: 
2023-03-29T03:43:39.5185041Z         	Error Trace:	D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2415
2023-03-29T03:43:39.5186163Z         	            				D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2429
2023-03-29T03:43:39.5186465Z         	Error:      	Condition never satisfied
2023-03-29T03:43:39.5186880Z         	Test:       	TestWSClient_SubscriptionsCompat/deprecated,_filtered
2023-03-29T03:43:39.5187074Z === CONT  TestWSClient_SubscriptionsCompat
2023-03-29T03:43:39.5187659Z     logger.go:130: 2023-03-29T03:35:04.261Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[0]"}
2023-03-29T03:43:39.5188230Z     logger.go:130: 2023-03-29T03:35:04.749Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[1]"}
2023-03-29T03:43:39.5188792Z     logger.go:130: 2023-03-29T03:35:06.082Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[2]"}
2023-03-29T03:43:39.5189531Z     logger.go:130: 2023-03-29T03:35:06.901Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[3]"}
2023-03-29T03:43:39.5189904Z === RUN   TestWSClient_SubscriptionsCompat/deprecated,_non-filtered
2023-03-29T03:43:39.5191138Z === CONT  TestWSClient_SubscriptionsCompat
2023-03-29T03:43:39.5192054Z     logger.go:130: 2023-03-29T03:35:08.756Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[block_added]"}
2023-03-29T03:43:39.5192874Z     logger.go:130: 2023-03-29T03:35:09.764Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[transaction_added]"}
2023-03-29T03:43:39.5193572Z     logger.go:130: 2023-03-29T03:35:10.375Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[notification_from_execution]"}
2023-03-29T03:43:39.5194255Z     logger.go:130: 2023-03-29T03:35:11.777Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[transaction_executed]"}
2023-03-29T03:43:39.5195271Z     logger.go:130: 2023-03-29T03:35:12.677Z	DEBUG	done processing headers	{"headerIndex": 2, "blockHeight": 1, "took": "191.1356ms"}
2023-03-29T03:43:39.5196170Z     logger.go:130: 2023-03-29T03:35:13.306Z	INFO	persisted to disk	{"blocks": 1, "keys": 44, "headerHeight": 2, "blockHeight": 2, "took": "0s"}
2023-03-29T03:43:39.5196553Z === CONT  TestWSClient_SubscriptionsCompat/deprecated,_non-filtered
2023-03-29T03:43:39.5196702Z     client_test.go:2415: 
2023-03-29T03:43:39.5197295Z         	Error Trace:	D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2415
2023-03-29T03:43:39.5198347Z         	            				D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2432
2023-03-29T03:43:39.5198885Z         	Error:      	Condition never satisfied
2023-03-29T03:43:39.5199592Z         	Test:       	TestWSClient_SubscriptionsCompat/deprecated,_non-filtered
2023-03-29T03:43:39.5199788Z === CONT  TestWSClient_SubscriptionsCompat
2023-03-29T03:43:39.5200339Z     logger.go:130: 2023-03-29T03:35:13.916Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[0]"}
2023-03-29T03:43:39.5201070Z     logger.go:130: 2023-03-29T03:35:14.733Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[1]"}
2023-03-29T03:43:39.5202691Z     logger.go:130: 2023-03-29T03:35:15.820Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[2]"}
2023-03-29T03:43:39.5203349Z     logger.go:130: 2023-03-29T03:35:16.599Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[3]"}
2023-03-29T03:43:39.5203585Z === RUN   TestWSClient_SubscriptionsCompat/relevant,_filtered
2023-03-29T03:43:39.5203777Z === CONT  TestWSClient_SubscriptionsCompat
2023-03-29T03:43:39.5204706Z     logger.go:130: 2023-03-29T03:35:17.399Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[block_added ]"}
2023-03-29T03:43:39.5205830Z     logger.go:130: 2023-03-29T03:35:18.213Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[transaction_added ]"}
2023-03-29T03:43:39.5206469Z     logger.go:130: 2023-03-29T03:35:19.051Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[notification_from_execution ]"}
2023-03-29T03:43:39.5207370Z     logger.go:130: 2023-03-29T03:35:20.250Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[transaction_executed ]"}
2023-03-29T03:43:39.5207992Z     logger.go:130: 2023-03-29T03:35:21.728Z	DEBUG	done processing headers	{"headerIndex": 3, "blockHeight": 2, "took": "626.7157ms"}
2023-03-29T03:43:39.5208825Z     logger.go:130: 2023-03-29T03:35:22.535Z	INFO	persisted to disk	{"blocks": 1, "keys": 33, "headerHeight": 3, "blockHeight": 3, "took": "511µs"}
2023-03-29T03:43:39.5209225Z === CONT  TestWSClient_SubscriptionsCompat/relevant,_filtered
2023-03-29T03:43:39.5209624Z     client_test.go:2512: 
2023-03-29T03:43:39.5210322Z         	Error Trace:	D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2512
2023-03-29T03:43:39.5211192Z         	            				D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2526
2023-03-29T03:43:39.5211469Z         	Error:      	Condition never satisfied
2023-03-29T03:43:39.5211848Z         	Test:       	TestWSClient_SubscriptionsCompat/relevant,_filtered
2023-03-29T03:43:39.5212029Z === CONT  TestWSClient_SubscriptionsCompat
2023-03-29T03:43:39.5212689Z     logger.go:130: 2023-03-29T03:35:23.348Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[0]"}
2023-03-29T03:43:39.5213264Z     logger.go:130: 2023-03-29T03:35:24.480Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[1]"}
2023-03-29T03:43:39.5213776Z     logger.go:130: 2023-03-29T03:35:25.231Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[2]"}
2023-03-29T03:43:39.5214293Z     logger.go:130: 2023-03-29T03:35:26.374Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[3]"}
2023-03-29T03:43:39.5214633Z === RUN   TestWSClient_SubscriptionsCompat/relevant,_non-filtered
2023-03-29T03:43:39.5214956Z === CONT  TestWSClient_SubscriptionsCompat
2023-03-29T03:43:39.5215526Z     logger.go:130: 2023-03-29T03:35:27.368Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[block_added]"}
2023-03-29T03:43:39.5216093Z     logger.go:130: 2023-03-29T03:35:28.272Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[transaction_added]"}
2023-03-29T03:43:39.5216702Z     logger.go:130: 2023-03-29T03:35:29.094Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[notification_from_execution]"}
2023-03-29T03:43:39.5217281Z     logger.go:130: 2023-03-29T03:35:30.059Z	DEBUG	processing rpc request	{"method": "subscribe", "params": "[transaction_executed]"}
2023-03-29T03:43:39.5217834Z     logger.go:130: 2023-03-29T03:35:31.922Z	DEBUG	done processing headers	{"headerIndex": 4, "blockHeight": 3, "took": "0s"}
2023-03-29T03:43:39.5218361Z     logger.go:130: 2023-03-29T03:35:32.887Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[0]"}
2023-03-29T03:43:39.5218971Z     logger.go:130: 2023-03-29T03:35:32.947Z	INFO	persisted to disk	{"blocks": 1, "keys": 40, "headerHeight": 4, "blockHeight": 4, "took": "0s"}
2023-03-29T03:43:39.5219498Z     logger.go:130: 2023-03-29T03:35:33.287Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[1]"}
2023-03-29T03:43:39.5220126Z     logger.go:130: 2023-03-29T03:35:34.100Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[2]"}
2023-03-29T03:43:39.5220698Z     logger.go:130: 2023-03-29T03:35:35.291Z	DEBUG	processing rpc request	{"method": "unsubscribe", "params": "[3]"}
2023-03-29T03:43:39.5221197Z     logger.go:130: 2023-03-29T03:35:35.923Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:51350"}
2023-03-29T03:43:39.5221439Z --- FAIL: TestWSClient_SubscriptionsCompat (46.85s)
2023-03-29T03:43:39.5222010Z     --- FAIL: TestWSClient_SubscriptionsCompat/deprecated,_filtered (13.74s)
2023-03-29T03:43:39.5222441Z     --- FAIL: TestWSClient_SubscriptionsCompat/deprecated,_non-filtered (8.69s)
2023-03-29T03:43:39.5222859Z     --- FAIL: TestWSClient_SubscriptionsCompat/relevant,_filtered (9.84s)
2023-03-29T03:43:39.5223279Z     --- PASS: TestWSClient_SubscriptionsCompat/relevant,_non-filtered (8.88s)
```

--- a/pkg/services/rpcsrv/client_test.go
+++ b/pkg/services/rpcsrv/client_test.go
@@ -2124,19 +2124,20 @@ func TestWSClient_SubscriptionsCompat(t *testing.T) {
 	blocks := getTestBlocks(t)
 	bCount := uint32(0)
 
-	getData := func(t *testing.T) (*block.Block, byte, util.Uint160, string, string) {
+	getData := func(t *testing.T) (*block.Block, *block.Block, byte, util.Uint160, string, string) {
 		b1 := blocks[bCount]
 		primary := b1.PrimaryIndex
 		tx := b1.Transactions[0]
 		sender := tx.Sender()
 		ntfName := "Transfer"
 		st := vmstate.Halt.String()
-		bCount++
-		return b1, primary, sender, ntfName, st
+		b2 := blocks[bCount+1]
+		bCount += 2
+		return b1, b2, primary, sender, ntfName, st
 	}
 
 	checkRelevant := func(t *testing.T, filtered bool) {
-		b, primary, sender, ntfName, st := getData(t)
+		b, bNext, primary, sender, ntfName, st := getData(t)
 		var (
 			bID, txID, ntfID, aerID string
 			blockCh                 = make(chan *block.Block)
@@ -2212,6 +2213,10 @@ func TestWSClient_SubscriptionsCompat(t *testing.T) {
 
 		// Accept the next block and wait for events.
 		require.NoError(t, chain.AddBlock(b))
+		// Blockchain's events channel is not buffered, and thus, by adding one more extra block
+		// we're ensuring that the previous block event receiving was successfully handled by Blockchain's
+		// notificationDispatcher loop. Once we're sure in that, we may start to check the actual notifications.
+		require.NoError(t, chain.AddBlock(bNext))
 		assert.Eventually(t, func() bool {
 			lock.RLock()
 			defer lock.RUnlock()

